vendor/api/v1/stripe/views.py

```diff
@@ -224,15 +230,15 @@ def post(self, request, *args, **kwargs):
         try:
             subscription = Subscription.objects.get(gateway_id=stripe_invoice.subscription, profile=customer_profile)
         except ObjectDoesNotExist:
-            logger.error(f"StripeSubscriptionInvoicePaid customer: {customer_profile} does not have a subscription with stripe_id: {stripe_invoice.subscription}")
-            return HttpResponse(status=400)
+            logger.error(f"StripeInvoicePaid customer: {customer_profile} does not have a subscription with stripe_id: {stripe_invoice.subscription}")
+            return HttpResponse(status=400, content=f"StripeInvoicePaid customer: {customer_profile} does not have a subscription with stripe_id: {stripe_invoice.subscription}")
 
-        payment = subscription.payments.filter(transaction=None, status=PurchaseStatus.QUEUED).first()
-        receipt = subscription.receipts.filter(transaction=None).first()
+        payment = subscription.payments.filter(Q(transaction="") | Q(transaction=None), status=PurchaseStatus.QUEUED).first()
```

Comment:
Im sure you've tested, but since you're combining AND and OR would something like this be more readable?

```suggestion
        payment = subscription.payments.filter(status=PurchaseStatus.QUEUED).filter(Q(transaction="") | Q(transaction=None)).first()
```
or

```suggestion
        payment = subscription.payments.filter((Q(transaction="") | Q(transaction=None)), status=PurchaseStatus.QUEUED).first()
```